easybuild/framework/easyblock.py

```diff
@@ -947,6 +947,10 @@ def make_module_dep(self, unload_info=None):
         self.log.debug("List of excluded deps: %s", excluded_deps)
 
         deps = [d for d in deps if d not in excluded_deps]
+        for dep in excluded_deps:
+            excluded_dep_deps = dependencies_for(dep, self.modules_tool)
```

Comment:
@bartoldeman That seems like a nice way to introduce a hard-to-debug situation that would only pop up in certain situations...

I'm OK with making sure that all modules relevant for the toolchain being used are made available in this context, and then cleaning up afterwards (we already do exactly this in other places too), it just makes things a bit more complex (and slow).

Thoughts @ocaisa?